setup/kubernetes/aws.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Set Up an Amazon Elastic Kubernetes Cluster
+description: Learn how to set up an Amazon EKS cluster for your Coder deployment.
+---
+
+This deployment guide shows you how to set up an Amazon Elastic Kubernetes
+Engine cluster on which Coder can deploy.
+
+## Prerequisites
+
+Before proceeding, please make sure that you have the [eksctl command line
+utility](https://docs.aws.amazon.com/eks/latest/userguide/getting-started-eksctl.html)
+and the [AWS Command Line
+Interface](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html)
+installed on your machine.
+
+## Step 1: Spin up a K8 Cluster
+
+The following will spin up a Kubernetes cluster using the eksctl command;
+replace the parameters and environment variables as needed to reflect those for
+your environment.
+
+```bash
+CLUSTER_NAME="MY_CLUSTER_NAME" \
+  SSH_KEY_PATH="~/.ssh/my-public-key.pub" REGION="us-east-2" \
+  eksctl create cluster \
+  --name "$CLUSTER_NAME" \
+  --version 1.17 \
+  --region "$REGION" \
+  --nodegroup-name standard-workers \
+  --node-type t3.medium \
+  --nodes 2 \
+  --nodes-min 2 \
+  --nodes-max 8 \
+  --ssh-access \
+  --ssh-public-key "$SSH_KEY_PATH" \
+  --managed
+```
+
+## Step 2: Adjust the K8 Storage Class
+
+Once you've created the cluster, adjust the default Kubernetes storage class to
+support immediate volume binding.
+
+First, delete the gp2 storage class: `kubectl delete sc gp2`
```

Comment:
We actually should ensure that `kubectl` is pointed to the right context first just in case `eksctl` didn't set the context automatically. Right now `eksctl` does that, but if it changes in the future this could delete the wrong storage class.

`kubectl config current-context` which should return with the cluster you created.